web/web.go

```diff
@@ -240,9 +241,11 @@ func MustBuildServerURL(pathPrefix string) string {
 	if err != nil {
 		panic(err)
 	}
-	hostname, err := os.Hostname()
-	if err != nil {
-		panic(err)
+	if hostname == "" {
+		hostname, err = os.Hostname()
+		if err != nil {
+			panic(err)
+		}
 	}
 	return fmt.Sprintf("http://%s:%s%s", hostname, port, pathPrefix)
```

Comment:
I'd keep the `globalURL` function as it is, except for also using the `web.hostname` flag value instead of `os.Hostname()`. The `globalURL` function is used only for the actual clickable link of a target on the /status page of the Prometheus server. So that when Prometheus monitors itself or something else at localhost, the linkified version will still take you to the right place when looking at the status page remotely. We can't generally tell whether that global URL should also contain the path prefix or not - it would depend on whether the Prometheus is scraping something with a path prefix (like itself) or not... so I'd say just keep it as it is and just change how it gets its hostname.